pkg/controller/logstash/reconcile.go

```diff
@@ -41,11 +48,63 @@ func reconcileStatefulSet(params Params, podTemplate corev1.PodTemplateSpec) (*r
 		RevisionHistoryLimit: params.Logstash.Spec.RevisionHistoryLimit,
 		VolumeClaimTemplates: params.Logstash.Spec.VolumeClaimTemplates,
 	})
-	if err := controllerutil.SetControllerReference(&params.Logstash, &s, scheme.Scheme); err != nil {
+
+	recreations := 0
+
+	sSetRecreations, err := volume.RecreateStatefulSets(params.Context, params.Client, &params.Logstash)
+
+	recreations += sSetRecreations
+	if err != nil {
+		if apierrors.IsConflict(err) {
+			ulog.FromContext(params.Context).V(1).Info("Conflict while recreating stateful set, requeueing", "message", err)
+			return results.WithResult(reconcile.Result{Requeue: true}), params.Status
+		}
+		return results.WithError(fmt.Errorf("StatefulSet recreation: %w", err)), params.Status
+	}
+
+	if recreations > 0 {
+		// Some StatefulSets are in the process of being recreated to handle PVC expansion:
+		// it is safer to requeue until the re-creation is done.
+		// Otherwise, some operation could be performed with wrong assumptions:
+		// the sset doesn't exist (was just deleted), but the Pods do actually exist.
+		ulog.FromContext(params.Context).V(1).Info("StatefulSets recreation in progress, re-queueing after 30 seconds.", "namespace", params.Logstash.Namespace, "ls_name", params.Logstash.Name,
+			"recreations", recreations, "status", params.Status)
+		return results.WithResult(reconcile.Result{RequeueAfter: 30 * time.Second}), params.Status
+	}
+	ok, _, err := params.expectationsSatisfied(params.Context)
```

Comment:
Do you think this checking can do before `volume.RecreateStatefulSets()` like [here](https://github.com/elastic/cloud-on-k8s/blob/main/pkg/controller/elasticsearch/driver/nodes.go#L55) to quit earlier when expectations do not match ?